index.js

```diff
@@ -62,7 +120,192 @@ function createTaskCreatedAt(taskCreatedAt) {
 function createTaskCheckBox() {
   return `<input
             type="checkbox"
-            class="table-box-check-box"
             name="checkbox-done"
           /> `;
 }
+
+function sortTasks(typeOfSort) {
+  if (typeOfSort == "a-to-z") {
+    sortTasksAtoZ();
+  } else {
+    unsortTasks();
+  }
+}
+
+function sortTasksAtoZ() {
+  const collator = new Intl.Collator("en", {
+    numeric: true,
+  });
+
+  tasks = tasks.sort((randomTask1, randomTask2) =>
+    collator.compare(
+      randomTask1.title.toUpperCase(),
+      randomTask2.title.toUpperCase()
+    )
+  );
+
+  renderTasks();
+}
+
+function unsortTasks() {
+  tasks.sort((randomTask1, randomTask2) => {
+    const randomTask1ID = randomTask1.id;
+    const randomTask2ID = randomTask2.id;
+    let comparisonResult = 0;
+    if (randomTask1ID > randomTask2ID) {
+      comparisonResult = 1;
+    } else if (randomTask1ID < randomTask2ID) {
+      comparisonResult = -1;
+    }
+    return comparisonResult;
+  });
+
+  renderTasks();
+}
+
+function searchTitle(text) {
+  if (text != null || "") {
+    tasks = tasks.filter((task) => task.title.includes(text));
+    renderTasks();
+    tasks = allTasks;
+  } else {
+    renderTasks();
+  }
+}
+
+function filterTasksByTime(time) {
+  const peresentTime = new Date().toISOString();
+
+  if (time == "no-exact-time") {
+    renderTasks();
+  }
+  if (time == "today") {
+    filterTodayTasks(peresentTime);
+  } else if (time == "last-day") {
+    filterLastDayTasks(peresentTime);
+  } else if (time == "last-7-days") {
+    filterLast7DaysTasks(peresentTime);
+  } else if (time == "last-30-days") {
+    filterLast30DaysTasks(peresentTime);
```

Comment:
```js
Date.now() - 1 * 24 * 60 * 60 * 1000 ; // one day before (timestamp)
Date.now() - 7 * 24 * 60 * 60 * 1000 ; // 7 days before (timestamp)
Date.now() - 30 * 24 * 60 * 60 * 1000 ; // 30 days before (timestamp)
```
Then you can have:
```js
new Date("2022-09-02T11:45:05.267Z").getTime()
```
And simply compare the results as a number. So, sudo code for filter:
```js
return new Date(task.createdAt).getTime() > Date.now() - 1 * 24 * 60 * 60 * 1000; // tasks of last day
```

Reply:
to compare **LastDays** tasks ,is it important to setHours(0,0,0,0) ?

Reply:
That was a big help for me , I search about **Date** again and leaned so many things.